features/step_definitions/matching.js

```diff
@@ -121,41 +134,46 @@ module.exports = function () {
                     var q = d3.queue();
 
                     var testSubMatching = (sub, si, scb) => {
-                        if (si >= subMatchings.length) {
-                            ok = false;
-                            q.abort();
-                            scb();
-                        } else {
-                            var sq = d3.queue();
-
-                            var testSubNode = (ni, ncb) => {
-                                var node = this.findNodeByName(sub[ni]),
-                                    outNode = subMatchings[si][ni];
-
-                                if (this.FuzzyMatch.matchLocation(outNode, node)) {
-                                    encodedResult += sub[ni];
-                                    extendedTarget += sub[ni];
-                                } else {
+                        var sq = d3.queue();
+                        var testSubNode = (ni, ncb) => {
```

Comment:
Same for this helper function. Nothing that gets called is asynchronous. You can omit the callback and instead return the result directly.